include/swift/Syntax/Syntax.h

```diff
@@ -194,8 +328,196 @@ class Syntax {
   AbsoluteOffsetPosition getAbsoluteEndPositionAfterTrailingTrivia() const {
     return Data->getAbsoluteEndPositionAfterTrailingTrivia();
   }
+};
+
+// MARK: - SyntaxRef
+
+/// See comment on top of file.
+class SyntaxRef {
+  const SyntaxDataRef *Data;
```

Comment:
I think the pointer itself is also never changed. So `const SyntaxDataRef * const Data`

Reply:
👍 Also changed `Data` in `Syntax` to `const RC<const SyntaxData> Data`.